### PR TITLE
All the enhancements I had to do quickly for the demo

### PR DESCRIPTION
**Pull request by goldbergyoni**
I don't use to commit big PR, this time I have to because during the demo I had to change many things quickly

### CODE REVIEW

**Review by DanielGluskin — APPROVED** [review]

File: various-receipes/component-isolation/test/anti-pattern.isolation.test.js [hunk 1]
```diff
@@ -37,15 +41,16 @@ afterAll(async (done) => {
...
       //Arrange
       process.env.SEND_MAILS = 'true';
-      sinon
-        .stub(OrderRepository.prototype, 'addOrder')
-        .throws(new Error('Unknown error'));
+
       // ❌ Anti-Pattern: The call will succeed regardless if the input, even if no mail address will get provided
       // We're not really simulating the integration data
-      const emailHTTPCall = nock('https://mailer.com').post('/send').reply(202);
+      const emailHTTPCall = nock('http://localhost')
```
Comment (mikicho) [inline various-receipes/component-isolation/test/anti-pattern.isolation.test.js:50]: please change this to `mailer.com`

File: various-receipes/component-isolation/test/isolation.test.js [hunk 2]
```diff
@@ -36,14 +37,51 @@ afterEach(() => {
...
   await stopWebServer();
+
+  // ️️️✅ Best Practice: Clean-up all nocks before the next file starts
   nock.enableNetConnect();
   done();
 });
 
 // ️️️✅ Best Practice: Structure tests
-describe('/api', () => {
+describe.skip('/api', () => {
```
Comment (mikicho) [inline various-receipes/component-isolation/test/isolation.test.js:47]: skip?

File: various-receipes/component-isolation/test/isolation.test.js [hunk 3]
```diff
@@ -61,7 +99,7 @@ describe('/api', () => {
       expect(orderAddResult.status).toBe(200);
     });
 
-    test('When the user does not exist, return http 404', async () => {
+    tests.skip('When the user does not exist, return http 404', async () => {
```
Comment (mikicho) [inline various-receipes/component-isolation/test/isolation.test.js:102]: skip?

File: various-receipes/data-isolation/anti-pattern-data-isolation.test.js [hunk 4]
```diff
@@ -1,41 +1,40 @@
...
 
-  const orderToAdd = {
-    userId: 1,
-    productId: 2,
-    mode: "approved",
-  };
-
   // ❌ Anti-Pattern: Adding global records which are mutated by the tests. This will lead to high coupling and flakiness
-  existingOrderId = (await request(expressApp).post("/order").send(orderToAdd)).body.id;
+  //existingOrderId = (await request(expressApp).post("/order").send({userId:1, mode:'approved'})).body.id;
```
Comment (mikicho) [inline various-receipes/data-isolation/anti-pattern-data-isolation.test.js:19]: why comment?

File: various-receipes/data-isolation/anti-pattern-data-isolation.test.js [hunk 5]
```diff
@@ -1,41 +1,40 @@
...
 });
 
-afterEach(() => {
+afterEach(async () => {
   nock.cleanAll();
   sinon.restore();
 
   // ❌ Anti-Pattern: Cleaning here now will affect tests in other processes
-  await new OrderRepository().cleanup();
+  //await new OrderRepository().cleanup();
```
Comment (mikicho) [inline various-receipes/data-isolation/anti-pattern-data-isolation.test.js:37]: why comment?

File: various-receipes/friendly-structure-for-reporting/jest.config.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  verbose: true,
+  //testMatch: ['**/anti-pattern-flat-report.test.js'],
```
Comment (mikicho) [inline various-receipes/friendly-structure-for-reporting/jest.config.js:3]: why comment?

**Review by mikicho — CHANGES_REQUESTED** [review]
Great stuff! 🥇 
1. why we need this Docx file?
2. it was hard to review it with all these prettier changes. 
3. Do we need all those empty test files?

File: various-receipes/friendly-structure-for-reporting/jest.config.js (hunk 6 above)
Comment (goldbergyoni) [reply various-receipes/friendly-structure-for-reporting/jest.config.js:3]: @mikicho I don't want to run anti-patterns files

**Comment by mikicho** [comment]
@goldbergyoni great, what about this docx file?

**Comment by goldbergyoni** [comment]
> @goldbergyoni great, what about this docx file? do we need it?

Yes, it will be used to show how a good component/integration report is a mirror of the requirement

@mikicho

File: various-receipes/doc-driven-contract-test/test/contract-example-with-openapi.test.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 const path = require('path');
+const axios = require('axios');
```
Comment (mikicho) [inline various-receipes/doc-driven-contract-test/test/contract-example-with-openapi.test.js]: I think this is redundant

**Review by mikicho — APPROVED** [review]
🐤

**Pull request merged**